pkg/agent/sasl.go

```diff
@@ -0,0 +1,39 @@
+package agent
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/segmentio/kafka-go/sasl"
+	"github.com/segmentio/kafka-go/sasl/plain"
+	"github.com/segmentio/kafka-go/sasl/scram"
+)
+
+func buildSASLConfig(cfg *Config) (sasl.Mechanism, error) {
+	// Read client ID
+	id, err := os.ReadFile(cfg.KafkaSASLClientIDPath)
+	if err != nil {
+		return nil, err
+	}
+	strID := strings.TrimSpace(string(id))
+	// Read password
+	pwd, err := os.ReadFile(cfg.KafkaSASLClientSecretPath)
+	if err != nil {
+		return nil, err
+	}
+	strPwd := strings.TrimSpace(string(pwd))
+	var mechanism sasl.Mechanism
+	switch cfg.KafkaSASLType {
+	case "plain":
+		mechanism = plain.Mechanism{Username: strID, Password: strPwd}
+	case "scramSHA512":
+		mechanism, err = scram.Mechanism(scram.SHA512, strID, strPwd)
+	default:
+		return nil, fmt.Errorf("unknown SASL type: %s", cfg.KafkaSASLType)
+	}
+	if err != nil {
```

Comment:
there is no need for this block since u already return on err

Reply:
actually, the line above was setting an error; I modified this part so it's more visible that an error might be produced